src/main/scala/io/pinecone/spark/pinecone/package.scala

```diff
@@ -3,20 +3,23 @@ package io.pinecone.spark
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.google.protobuf.{ListValue, Struct, Value}
 import org.apache.spark.sql.connector.write.WriterCommitMessage
-import org.apache.spark.sql.types.{ArrayType, FloatType, IntegerType, StringType, StructType}
+import org.apache.spark.sql.types.{ArrayType, FloatType, IntegerType, StringType, StructField, StructType}
 
 import scala.collection.JavaConverters._
 
 package object pinecone {
-  // TODO: Fill out the schema (mainly solve the whole metadata issue)
   val COMMON_SCHEMA: StructType =
     new StructType()
-      .add("id", StringType)
-      .add("namespace", StringType)
-      .add("values", ArrayType(FloatType))
-      .add("metadata", StringType)
-      .add("sparse_id", ArrayType(IntegerType, containsNull = true), nullable = true)
-      .add("sparse_values", ArrayType(FloatType, containsNull = true), nullable = true)
+      .add("id", StringType, nullable = false)
+      .add("namespace", StringType, nullable = true)
+      .add("values", ArrayType(FloatType, containsNull = false), nullable = false)
+      .add("metadata", StringType, nullable = true)
+      .add("sparse_values", StructType(
+        Array(
+          StructField("indices", ArrayType(IntegerType, containsNull = true), nullable = true),
+          StructField("values", ArrayType(FloatType, containsNull = true), nullable = true)
```

Comment:
I could be wrong about how `indices` and `values` need to map, but is it possible for these arrays to contain `null`? I was assuming it would look something like:
```suggestion
          StructField("indices", ArrayType(IntegerType, containsNull = false), nullable = true),
          StructField("values", ArrayType(FloatType, containsNull = false), nullable = true)
```

Reply:
Okay so after experimenting with the api and based on api spec, this is now the updated behavior:

If sparse_values is defined, then: 
1. the indices and values must be defined for the node (unit test #3, #4, and #5)
2. No nulls can be passed (unit test #6 and #7)

If sparse_values is not defined, then the user should still be able to send the dense vector (BasicIntegrationTest).

Also BasicIntegrationTest takes care of:
1. id and values are defined only
2. id and values along with all optional fields
3. id, values, and sparse_values with indices and values
4. id, values, and empty ("") namespace
5. id, values, and metadata only
6. and 2 more repetition with other fields and extra metadata